packages/ui/src/components/Table/STable.vue

```diff
@@ -4,24 +4,27 @@ import { MaybeElementRef, not } from '@vueuse/core'
 import { get, findLast } from 'lodash-es'
 import { SCheckboxAtom } from '@/components/Checkbox'
 import { IconArrowTop16, IconArrowsChevronDownRounded24 } from '@/components/icons'
-import { useColumnSort } from './column-sort.composable'
+import { useColumnSort } from './use-column-sort'
 import {
   TableRow,
-  CellEventData,
-  HeaderEventData,
-  RowEventData,
-  SortEventData,
-  ColumnSortOrder,
+  TableCellEventData,
+  TableHeaderEventData,
+  TableRowEventData,
+  TableSortEventData,
+  TableColumnSortOrder,
   TableCellConfigCallbackParams,
   TableRowConfigCallbackParams,
   TableHeaderCellConfigCallbackParams,
 } from './types'
-import { useFlexColumns } from './flex-columns-widths.composable'
-import { useRowSelect } from './row-select.composable'
-import { useColumnExpand } from './column-expand.composable'
+import { useFlexColumns } from './use-flex-columns-widths'
+import { useRowSelect } from './use-row-select'
+import { useColumnExpand } from './use-column-expand'
 import { isDefaultColumn, isExpandColumn, isRecord, isSelectionColumn } from './utils'
-import { ActionColumnApi, ColumnApi, TABLE_API_KEY } from './api'
-import { useTableHeights } from './table-heights.composable'
+import { TableActionColumnApi, TableColumnApi, TABLE_API_KEY } from './api'
+import { useTableHeights } from './use-table-heights'
+
+// vue can't infer Object prop type from CSSProperties and logs warnings and this helps
+type CSSObject = Partial<CSSProperties>
```

Comment:
In this case we can use plain props object with `PropType<T>` typing. This allows us to use the most flexible way to define props.